src/common/transformations/src/transformations/common_optimizations/transpose_sinking_utils.cpp

```diff
@@ -146,14 +195,21 @@ NodeVector InsertOutputTransposes(NodePtr main_node, TransposeInputsInfo& transp
 }  // namespace sink_forward
 
 namespace sink_backward {
+
 NodeVector InsertTransposeBeforeNode(NodePtr main_node, std::shared_ptr<Constant> transpose_const) {
     const auto transpose_axis_order = transpose_const->get_axis_vector_val();
     const auto transpose_element_type = transpose_const->get_element_type();
 
+    if (HasDynamicRankInput(main_node))
+        return {};
+
     NodeVector new_nodes;
 
+    const auto max_input_rank = GetMaxInputRank(main_node);
```

Comment:
it looks like we need to check that `max_input_rank` is not `-1`

Reply:
Thank you. I check it in UpdateInputTransposes but here I forgot